File: 6_Kyu_Kata/word_a10n_abbreviation/JavaScript/wordA10nAbbreviation.js
// 6 kyu - Word a10n (abbreviation)  [ ID: 5375f921003bf62192000746  (word-a10n-abbreviation) ]
// URL: https://www.codewars.com/kata/5375f921003bf62192000746
// Category: undefined  |  Tags: STRINGS | FUNDAMENTALS
// *****************************************************************************
const abbreviate = (s) => s.replace(/(\w)\w{2,}(\w)/g, (v, $1, $2) => $1 + (v.length - 2) + $2)

File: 6-kyu-kata/word-a10n-abbreviation/python/word_a10n_abbreviation.py
# 6 kyu - Word a10n (abbreviation)  [ ID: 5375f921003bf62192000746  (word-a10n-abbreviation) ]
# URL: https://www.codewars.com/kata/5375f921003bf62192000746
# Category: REFERENCE  |  Tags: STRINGS | FUNDAMENTALS
# ******************************************************************************
import re
  
  
  def abbreviate(s):
      a8e = re.compile("(?<=[a-zA-Z])([a-zA-Z]{2,})(?=[a-zA-Z])")
      return re.sub(a8e, lambda x: str(len(x.group(0))), s)
